From d70da04905d284f9b5423bf297830451c62a777b Mon Sep 17 00:00:00 2001
From: root <root@linuxdev2.toradex.int>
Date: Sun, 2 Feb 2020 19:58:32 +0100
Subject: [PATCH] u-boot: tk1: make it build with python3 only installation

- Build libfdt with the python3 binary rather than using the inexistent
  python2 binary.
- Fallback to the hardcoded U-Boot assembly rather than using the
  python2 only BINMAN functionality.

Signed-off-by: root <root@linuxdev2.toradex.int>
---
 arch/arm/mach-tegra/Kconfig   | 1 -
 scripts/dtc/pylibfdt/Makefile | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/mach-tegra/Kconfig b/arch/arm/mach-tegra/Kconfig
index 97e22ead59..9f49e87829 100644
--- a/arch/arm/mach-tegra/Kconfig
+++ b/arch/arm/mach-tegra/Kconfig
@@ -41,7 +41,6 @@ config TEGRA_PMC_SECURE
 
 config TEGRA_COMMON
 	bool "Tegra common options"
-	select BINMAN
 	select BOARD_EARLY_INIT_F
 	select CLK
 	select DM
diff --git a/scripts/dtc/pylibfdt/Makefile b/scripts/dtc/pylibfdt/Makefile
index 15e66ad44d..42342c75bb 100644
--- a/scripts/dtc/pylibfdt/Makefile
+++ b/scripts/dtc/pylibfdt/Makefile
@@ -21,7 +21,7 @@ quiet_cmd_pymod = PYMOD   $@
 		CPPFLAGS="$(HOSTCFLAGS) -I$(LIBFDT_srcdir)" OBJDIR=$(obj) \
 		SOURCES="$(PYLIBFDT_srcs)" \
 		SWIG_OPTS="-I$(LIBFDT_srcdir) -I$(LIBFDT_srcdir)/.." \
-		$(PYTHON2) $< --quiet build_ext --inplace
+		$(PYTHON3) $< --quiet build_ext --inplace
 
 $(obj)/_libfdt.so: $(src)/setup.py $(PYLIBFDT_srcs) FORCE
 	$(call if_changed,pymod)
-- 
2.20.1

